Incorporate reviewers feedback on testing chapter (#101)
tidy-survey-r/tidy-survey-book: 06-statistical-testing.Rmd

The general idea of statistical testing is the same for data obtained through surveys and data obtained through other methods, where we compare the point estimates and variance estimates of each statistic to see if statistically significant differences exist. However, statistical testing for complex surveys involves additional considerations due to the need to account for the sampling design in order to obtain accurate variance estimates.

> Statistical testing, also called hypothesis testing, involves declaring a null and alternative hypothesis. A null hypothesis is denoted as $H_0$ and the alternative hypothesis is denoted as $H_A$. The null hypothesis is the "business as usual" situation while the alternative hypothesis is what we are trying to determine if the data supports. Stepping away from surveys, if we are rolling a dice, a null hypothesis would be that the die is fair and that each side has an equal chance of being rolled whereas the alternative hypothesis would be that the die is unfair. When we do hypothesis testing, we obtain a p-value and this shows how likely we are to observe the data if the null hypothesis is true. If the p-value (a probability between 0 and 1) is small, we have strong evidence to reject the null hypothesis as it is unlikely to see the data we are observing if the null hypothesis is true. However, if the p-value is large, we say we do not have evidence to reject the null hypothesis. We perform tests to say something about the population value after observing a sample.

Reviewer: ```suggestion
Statistical testing, also called hypothesis testing, involves declaring a null and alternative hypothesis. A null hypothesis is denoted as $H_0$ and the alternative hypothesis is denoted as $H_A$. The null hypothesis is the default assumption in that there are no differences in the data, or that the data is operating under "standard" behaviors.  On the other hand, the alternative hypothesis is the break from the "standard" and what we are trying to determine if the data supports. 

Let's review an example outside of survey data. If we are flipping a coin, a null hypothesis would be that the coin is fair and that each side has an equal chance of being flipped.  In other words, the probability of the coin landing on each side is 1/2.  Whereas an alternative hypothesis could be that the coin is unfair and that one side has a higher probability of being flipped (e.g., a probability of 1/4 to get heads, but a probability of 3/4 to get tails). We write this set of hypotheses as:

- $H_0: \rho_{heads} = \rho_{tails}$, where $\rho_{x}$ is the probability of flipping the coin and having it land on heads ($\rho_{heads}$) or tails ($\rho_{tails}$)
- $H_A: \rho_{heads} \neq \rho_{tails}$

When we conduct hypothesis testing, the statistical models calculate a p-value, which shows how likely we are to observe the data if the null hypothesis is true. If the p-value (a probability between 0 and 1) is small, we have strong evidence to reject the null hypothesis as it is unlikely to see the data we are observing if the null hypothesis is true. However, if the p-value is large, we say we do not have evidence to reject the null hypothesis. The size of the p-value for this cut off is determined by type 1 error known as $\alpha$.  A common type 1 error value for statistical testing is to use $\alpha = 0.05$.^[For more information on statistical testing, we recommend reviewing introduction to statistics textbooks.] It is common for explanations of statistical testing to refer to confidence level.  The confidence level is the inverse of the type 1 error.  Thus, if $\alpha = 0.05$, the confidence level would be 95%.
```

The functions in the {survey} package allow for the correct estimation of the variances. This chapter will cover the following statistical tests with survey data and functions:

* Comparison of proportions `svyttest()`
* Comparison of means `svyttest()`
* Goodness of fit tests `svygofchisq()`
* Tests of independence `svychisq()`
* Tests of homogeneity `svychisq()`

## Dot Notation {#dot-notation}

> Up to this point, we have shown functions that use wrappers from the {srvyr} package. This means that the functions work with tidyverse syntax. However, the functions in this chapter do not have wrappers in the {srvyr} package and are instead used directly from the {survey} package. Therefore, the design object is *not* the first argument, and to use these functions with the magrittr pipe `%>%` and tidyverse syntax, we will need to use dot (`.`) notation^[This could change in the future if another package is built or {srvyr} is expanded to work with **tidymodels** but no such plans are known at this time.]

Reviewer: ```suggestion
Up to this point, we have shown functions that use wrappers from the {srvyr} package. This means that the functions work with tidyverse syntax. However, the functions in this chapter do not have wrappers in the {srvyr} package and are instead used directly from the {survey} package. Therefore, the design object is *not* the first argument, and to use these functions with the magrittr pipe (`%>%`) and tidyverse syntax, we will need to use dot (`.`) notation^[This could change in the future if another package is built or {srvyr} is expanded to work with **tidymodels** but no such plans are known at this time.]
```

  - We need to use the survey design object instead of the original data frame
  - We can only use a formula and not separate x and y data

>   - The confidence level cannot be specified and will always be set to 95%. However, we will show examples of how the confidence level can be changed after running using the `confint()` function only after running the `svyttest()` function.

Reviewer: ```suggestion
  - The confidence level cannot be specified and will always be set to 95%. However, we will show examples of how the confidence level can be changed after running the `svyttest()` function by using the `confint()` function.
```

    p_{5_{Biden}} &= p_{5_{Trump}} = p_{5_{Other}},\\
    p_{6_{Biden}} &= p_{6_{Trump}} = p_{6_{Other}}
  \end{align*}

>     where $p_{i_{Biden}}$ is the observed proportion of each age group ($i$) that voted for President Joseph Biden, $p_{i_{Trump}}$ is the observed proportion of each age group ($i$) that voted for President Donald Trump, and $p_{i_{Other}}$ is the observed proportion of each age group ($i$) that voted for another candidate

Reviewer: ```suggestion
    where $p_{i_{Biden}}$ is the observed proportion of each age group ($i$) that voted for Joseph Biden, $p_{i_{Trump}}$ is the observed proportion of each age group ($i$) that voted for Donald Trump, and $p_{i_{Other}}$ is the observed proportion of each age group ($i$) that voted for another candidate
```

Reviewer: At the time of the voting, Biden wasn't president, so it would have been "Former Vice President Joseph Biden" and "President Donald Trump".  I think to avoid confusion we stick with just their names and not their titles.

(ref:stattest-chi-ex3-tab) Distribution of age group by presidential candidate selection with confidence intervals

```{r}
#| label: stattest-chi-ex3-tab
#| echo: FALSE
#| warning: FALSE

chi_ex3_obs_table %>%
    print_gt_book(knitr::opts_current$get()[["label"]])
```

> We can see that the age group distribution was younger for President Biden and other candidates and older for President Trump. For example, of those who voted for President Biden, 20.4% were in the 18-29 age group, compared to only 11.4% of those who voted for Trump were in that age group. On the other side, 23.4% of those who voted for Trump were in the 50-59 age group compared to only 15.4% of those who voted for President Biden.

Reviewer: ```suggestion
We can see that the age group distribution that voted for Biden and other candidates was younger than those that voted for Trump. For example, of those who voted for Biden, 20.4% were in the 18-29 age group, compared to only 11.4% of those who voted for Trump were in that age group. On the other side, 23.4% of those who voted for Trump were in the 50-59 age group compared to only 15.4% of those who voted for Biden.
```

ttest_solution2
```

> 3. Using the ANES data, does the average age (`Age`) of those who voted for President Biden in 2020 (`VotedPres2020_selection`) differ from those who voted for another candidate?

Reviewer: ```suggestion
3. Using the ANES data, does the average age (`Age`) of those who voted for Joseph Biden in 2020 (`VotedPres2020_selection`) differ from those who voted for another candidate?
```
